Tests/RxCocoaTests/SharedSequence+OperatorTest.swift

```diff
@@ -283,6 +290,31 @@ extension SharedSequenceOperatorTests {
         let expectedEvents = [1, 2, -1]
         XCTAssertEqual(events, expectedEvents)
     }
+    
+    func testAsDriver_doOnAfterNext() {
+        let hotObservable = BackgroundThreadPrimitiveHotObservable<Int>()
+        
+        var events = [Int]()
```

Comment:
This doesn't test the ordering properly.

Please turn this into `var events = [OrderedEvent]()`

````
struct OrderedEvent {
    case before(Event<Int>)
    case subscription(Event<Int>)
    case after(Event<Int>)
}

hotObservable.asDriver(onErrorJustReturn: -1).do(onAfterNext: { e in
            XCTAssertTrue(DispatchQueue.isMain)
            events.append(.after(.next(e)))
        })
        .do(onNext: { e in
                 events.push(.subscribe(.next(e)))
        }
````

Reply:
Done.